Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -511,14 +535,6 @@ private struct _BSONKeyedDecodingContainer<K: CodingKey>: KeyedDecodingContainer
         return entry
     }
 
-    /// Decode a BSONValue type from this container for the given key.
-    private func decodeBSONType<T: BSONValue>(_ type: T.Type, forKey key: Key) throws -> T {
-        let entry = try getValue(forKey: key)
-        return try self.decoder.with(pushedKey: key) {
```

Comment:
is still `with(pushedKey:)` thing still called on the new code paths that used to use this?

Reply:
what if you are decoding e.g. a `Bool` or a `String` (line 582, 597)? it seems to me like those can get called directly now without adding the key that is being decoded to the key path

Reply:
Ohh, I misunderstood what you were saying. Yeah that's a very good point; I shouldn't be using an "unboxing" method to do "decoding". I've gone ahead and reverted removing this method and made the two examples you mentioned call it again.